authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+
+As an alternative, you can xref:#sts-mode-installing-manual-config[create the required AWS resources manually].
+
+[id="sts-mode-installing-preflight-preparing"]
+==== Preparing the installer
+
+.Prerequisites
+* Obtain the pull secret for your cluster.
+* Obtain the {product-title} version 4.7 installation binary.
+
+.Procedure
+
+. From the directory that contains the installation program, obtain details of the {product-title} release image that your `openshift-install` binary is built to use:
++
+[source,terminal]
+----
+$ ./openshift-install version
+----
++
+.Example output
+[source,terminal]
+----
+./openshift-install 4.7.0
+built from commit 2d8b61dd8802058ffffcdc576e0cff67acebdb85
+release image registry.svc.ci.openshift.org/ocp/release@sha256:a322ab4941701452f428f5c138efb5354ad2aec33f6f7693f0a3033bfe257d8c
+----
+
+. Extract the AWS `CredentialsRequest` custom resources (CRs) from the release image:
++
+[source,terminal]
+----
+$ mkdir credreqs ; oc adm release extract --cloud=aws --credentials-requests quay.io/openshift-release-dev/ocp-release:4.<y>.<z>-x86_64 --to=./credreqs ; cat ./credreqs/*.yaml > credreqs.yaml
+----
++
+Where `<y>` and `<z>` are the numbers corresponding to the version of {product-title} you are installing.
+
+. Create the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ ./openshift-install create install-config
+----
+
+. Configure the cluster to install with the CCO in manual mode:
++
+[source,terminal]
+----
+$ echo "credentialsMode: Manual" >> install-config.yaml
+----
+
+. Create install manifests:
++
+[source,terminal]
+----
+$ ./openshift-install create manifests
+----
+
+[id="sts-mode-installing-preflight-using"]
+==== Using the sts-preflight tool
+
+.Prerequisites
+* Download the link:https://github.com/sjenning/sts-preflight[`sts-preflight` tool].
+
+.Procedure
+
+. Using an AWS account with administrator-level permissions, run the following command to:
++
+--
+** Generate public and private `ServiceAccount` object signing keys.
+** Create an S3 bucket.
+** Upload the OIDC config into the S3 bucket.
+** Set up an IAM Identity Provider that trusts the S3 bucket.
+** Create an IAM role with administrator rights to use for installation.
+** Create IAM roles for each AWS `CredentialsRequest` CR.
++
+[source,terminal]
+----
+$ ./sts-preflight create --infra-name <aws-infra-name> --region <aws-region> --credentials-requests-to-roles <path-to-credreqs.yaml>
+----
++
+Where `<aws-infra-name>` is a prefix the `sts-preflight` tool uses when naming the objects that are created in AWS.
+--
++
+The files that are required by the installer are added to the `_output` directory.
+
+. The `sts-preflight` tool does not put conditions on the IAM roles it creates to allow only specific `ServiceAccount` objects to assume a role. To restrict which identities can assume which role, you must manually modify the trust relationship of each role to add a corresponding condition.
```

Comment:
_tool does not put conditions on the IAM roles it creates to allow only specific `ServiceAccount` objects to assume a role_
Not clear to me what's going on here. Is this what we want or a negative consequence? 
_tool does not put conditions on the IAM roles it creates, **which allows** only specific `ServiceAccount` objects to assume a role_ ??

Reply:
Negative consequence. I flipped it around and added a complementizer - still a tricky sentence but I hope more clear?
_The `sts-preflight` tool does not put conditions that allow only specific `ServiceAccount` objects to assume a role on the IAM roles that it creates._